subworkflows/local/read_alignment_dna/main.nf

```diff
@@ -0,0 +1,218 @@
+            ch_inputs_normal_sorted.runnable.map { meta -> [meta, Utils.getNormalDnaSample(meta), 'normal'] },
+        )
+        .flatMap { meta, meta_sample, sample_type ->
+            meta_sample
+                .getAt(Constants.FileType.FASTQ)
+                .collect { key, fps ->
+                    def (library_id, lane) = key
+
+                    def meta_fastq = [
+                        key: meta.group_id,
+                        id: "${meta.group_id}_${meta_sample.sample_id}",
+                        sample_id: meta_sample.sample_id,
+                        library_id: library_id,
+                        lane: lane,
+                        sample_type: sample_type,
+                    ]
+
+                    return [meta_fastq, fps['fwd'], fps['rev']]
+                }
+        }
+
+    //
+    // MODULE: fastp
+    //
+    // Split FASTQ into chunks if requested for distributed processing
+    // channel: [ meta_fastq_ready, fastq_fwd, fastq_fwd ]
+    ch_fastqs_ready = Channel.empty()
+    if (max_fastq_records > 0) {
+
+        // Run process
+        FASTP(
+            ch_fastq_inputs,
+            max_fastq_records,
+        )
+
+        ch_versions = ch_versions.mix(FASTP.out.versions)
+
+        // Prepare outputs within conditional block
+        ch_fastqs_ready = FASTP.out.fastq
+            .flatMap { meta_fastq, reads_fwd, reads_rev ->
+
+                def data = [reads_fwd, reads_rev]
+                    .transpose()
+                    .collect { fwd, rev ->
+
+                        def split_fwd = fwd.name.replaceAll('\\..+$', '')
+                        def split_rev = rev.name.replaceAll('\\..+$', '')
+
+                        assert split_fwd == split_rev
+
+                        // NOTE(SW): split allows meta_fastq_ready to be unique, which is required during reunite below
+                        def meta_fastq_ready = [
+                            *:meta_fastq,
+                            id: "${meta_fastq.id}_${split_fwd}",
+                            split: split_fwd,
+                        ]
+
+                        return [meta_fastq_ready, fwd, rev]
+                    }
+
+                return data
+            }
+
+    } else {
+
+        ch_fastqs_ready = ch_fastq_inputs
+            .map { meta_fastq, fastq_fwd, fastq_rev ->
+
+                def meta_fastq_ready = [
+                    *:meta_fastq,
+                    split: null,
+                ]
+
+                return [meta_fastq_ready, fastq_fwd, fastq_rev]
+            }
+
+    }
+
+    //
+    // MODULE: BWA-MEM2
+    //
+    // Create process input channel
+    // channel: [ meta_bwamem2, fastq_fwd, fastq_rev ]
+    ch_bwamem2_inputs = ch_fastqs_ready
+        .map { meta_fastq_ready, fastq_fwd, fastq_rev ->
+
+            def meta_bwamem2 = [
+                *:meta_fastq_ready,
+
+
+                // TODO(SW): understand target format
```

Comment:
A cryptic TODO here

Reply:
- Removed in https://github.com/nf-core/oncoanalyser/pull/67